lib/contentful_model/query.rb

```diff
@@ -10,19 +10,131 @@ def <<(parameters)
       @parameters.merge!(parameters)
     end
 
+    def params(options)
+      self << options
+      self
+    end
+
+    def first
+      self << {'limit' => 1}
+      self.load.first
+    end
+
+    def offset(n)
+      self << {'skip' => n}
+      self
+    end
+    alias_method :skip, :offset
+
+    def limit(n)
+      self << {'limit' => n}
+      self
+    end
+
+    def locale(locale_code)
+      self << {'locale' => locale_code}
+      self
+    end
+
+    def paginate(page = 1, per_page = 100, order_field = 'sys.updatedAt')
+      page = 1 if page.nil? || !page.is_a?(Numeric) || page <= 0
+      per_page = 100 if per_page.nil? || !per_page.is_a?(Numeric) || per_page <= 0
+
+      skip_records_count = (page - 1) * per_page
+      self << { 'limit' => per_page, 'skip' => skip_records_count, 'order' => order_field }
+      self
+    end
+
+    def load_children(n)
+      self << {'include' => n}
+      self
+    end
+
+    def order(args)
+      prefix = ''
+      if args.is_a?(Hash)
+        column = args.first.first.to_s
+        prefix = '-' if args.first.last == :desc
+      elsif args.is_a?(Symbol)
+        column = args.to_s
+        prefix = ''
+      else
+        column = args.to_s
+      end
+      property_name = column.camelize(:lower).to_sym
+      sys_properties = ['type', 'id', 'space', 'contentType', 'linkType', 'revision', 'createdAt', 'updatedAt', 'locale']
+      property_type = sys_properties.include?(property_name.to_s) ? 'sys' : 'fields'
+
+      self << {'order' => "#{prefix}#{property_type}.#{property_name}"}
+      self
+    end
+
+    def find(id)
+      self << {'sys.id' => id}
+      self.load.first
+    end
+
+    def find_by(find_query = {})
+      find_query.each do |field, value|
+        key = if field.to_s.include?('sys.') || field.to_s.include?('fields.')
+                field
+              else
```

Comment:
does it make sense to add nice handling for `sys` properties as well so if a user does `find_by(id)` the query resolves to `sys.id=`?